sympy/combinatorics/coset_table.py

```diff
@@ -477,25 +511,49 @@ def coincidence(self, alpha, beta):
         A_dict_inv = self.A_dict_inv
         table = self.table
         p = self.p
-        l = 0
         # behaves as a queue
         q = []
-        self.merge(alpha, beta, q)
+        p_p = []
+        if modified:
+            p_p[beta] = self._grp.identity
+        l = 0
+        if modified:
+            self.modified_merge(alpha, beta, w, p_p, q)
+        else:
+            self.merge(alpha, beta, q)
         while len(q) > 0:
             gamma = q.pop(0)
             for x in A_dict:
                 delta = table[gamma][A_dict[x]]
                 if delta is not None:
                     table[delta][A_dict_inv[x]] = None
-                    mu = self.rep(gamma)
-                    nu = self.rep(delta)
+                    mu = None
+                    nu = None
+                    if modified:
+                        mu = self.modified_rep(gamma, p_p)
+                        nu = self.modified_rep(delta, p_p)
+                    else:
+                        mu = self.rep(gamma)
+                        nu = self.rep(delta)
```

Comment:
This is one example where having `p_p` as a `CosetTable` attribute would be useful: then you could just use `self.rep(gamma, modified=modified)`.